x-pack/plugins/maps/public/actions/visualize_geo_field_action.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import rison from 'rison-node';
+import uuid from 'uuid/v4';
+import { stringify } from 'query-string';
+import { i18n } from '@kbn/i18n';
+import { createAction } from '../../../../../src/plugins/ui_actions/public';
+import {
+  createKbnUrlStateStorage,
+  IKbnUrlStateStorage,
+} from '../../../../../src/plugins/kibana_utils/public';
+import { DiscoverAppState } from '../../../../../src/plugins/discover/public';
+import { getApplication, getIndexPatternService } from '../kibana_services';
+
+export const ACTION_VISUALIZE_GEO_FIELD = 'ACTION_VISUALIZE_GEO_FIELD';
+
+export const visualizeGeoFieldAction = createAction<typeof ACTION_VISUALIZE_GEO_FIELD>({
+  type: ACTION_VISUALIZE_GEO_FIELD,
+  getDisplayName: () =>
+    i18n.translate('xpack.maps.discover.visualizeFieldLabel', {
+      defaultMessage: 'Visualize on Kibana Maps',
+    }),
+  execute: async (context) => {
+    const indexPattern = await getIndexPatternService().get(context.indexPatternId);
+    const field = indexPattern.fields.find((fld) => fld.name === context.fieldName);
+    const supportsClustering = field?.aggregatable;
+    const stateStorage: IKbnUrlStateStorage = createKbnUrlStateStorage();
+    const appStateFromUrl: DiscoverAppState | null = stateStorage.get('_a');
```

Comment:
I wonder if we have a better way then this here. The problem would be, that this way this trigger needs to be emitted from Discover, and if it would be emitted somewhere else it would not work properly. I wonder if we can find a better way, e.g. the list of columns could be an additional parameter on the context for this action, like: `contextualFields`.

For (pinned) filters and query I wonder if we not anyway need/have a better way of transfering them via actions, since I assume this will be required for nearly all actions? cc @ppisljar @Dosant

Reply:
I will add the `columns` on the context. I could also add filter and query but if there is a better way to do it for all actions it will be great

Reply:
I agree we shouldn't rely on this info be present in url. 
Our existing similar actions get it from `context` 

For filters / query / timeRange you could just use query service from `data` plugin directly. 
e.g
`data.query.filterManger.getFilters()`
`data.query.queryString.getQuery()`

One way to inject is to wrap your action into function to create it:  
```
export const createVisualizeGeoFieldAction = (queryService) => createAction<typeof ACTION_VISUALIZE_GEO_FIELD>(
```

Reply:
Thanx @Dosant. Sounds good. Will look into this in order to remove the URL dependency